### ISSUE TITLE
[PATCH] Faster, simpler microdom.Element.__init__

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4437]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| spiv reported|
|-|-|
|Trac ID|trac#351|
|Type|enhancement|
|Created|2003-10-19 15:53:40Z|

Attachments:

* [microdom.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/002/0026476a20bfbd08714155bb66f0b4feb2d25c1c/a5a776971740d494a6d8512d691e9f5cb0f82d2b.patch) (1250 bytes) - added by spiv on 2003-10-19 15:53:40Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__351 351
type__enhancement enhancement
reporter__spiv spiv
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1066578820000000 1066578820000000
changetime__1468879015532746 1468879015532746
version__ 
owner__ 
cc__glyph cc__moonfallen cc__spiv
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
The attached patch removes a fairly pointless function and simplifies the code
slightly.  As a bonus, it speeds up Element.__init__ by 4% (as measured by
python2.3 /usr/lib/python2.3/timeit.py -s "from twisted.web.microdom import
Element" "Element('foo')").  The test suite still passes.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
moonfallen and glyph seem to be about equally to blame for microdom according to
"svn blame". Assigning to you though: c'mon, 14 lines!
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|moonfallen commented|
|-|-|

```
#!html
<pre>
Committed patch in r10867.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub commented|
|-|-|

In changeset 64704ca9ddef828761b43b4161a87755fcc720d1
```
#!CommitTicketReference repository="" revision="64704ca9ddef828761b43b4161a87755fcc720d1"
Merge pull request #351 from scop/8624-scop-close

Author: scop
Reviewer: rodrigc
Fixes: #8624

Close more files explicitly
```
